radixdlt/src/integration/java/com/radixdlt/consensus/simulation/checks/SyncOccurrenceCheck.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * (C) Copyright 2020 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.consensus.simulation.checks;
+
+import com.radixdlt.consensus.simulation.BFTCheck;
+import com.radixdlt.consensus.simulation.BFTSimulation;
+import com.radixdlt.counters.SystemCounters.CounterType;
+import io.reactivex.rxjava3.core.Completable;
+import io.reactivex.rxjava3.core.Observable;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+/**
+ * Periodically checks new occurences of syncing success
+ */
+public final class SyncOccurrenceCheck implements BFTCheck {
+	private final long time;
+	private final TimeUnit timeUnit;
+	private final Consumer<Long> assertion;
+
+	public SyncOccurrenceCheck(Consumer<Long> assertion, long time, TimeUnit timeUnit) {
```

Comment:
The meaning of the `Long` input to the `assertion` consumer is not obvious until you look at the implementation of `check`

Reply:
Yes, maybe "syncSuccessAssertion"?  Should also be LongConsumer.  Also documentation.  Also time -> duration.